assembly-h2o/build.gradle

```diff
@@ -0,0 +1,172 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+
+apply plugin: 'com.github.johnrengelman.shadow'
+apply plugin: 'base'
+
+description = """
+h2o and h2o Driver Assembly. This assembly extends provided jar by sparkling-water specific classes needed in external
+backend mode.
+
+User can specify location to either h2o or h2odriver jar via H2O_ORIGINAL_JAR env variable.
+No works is done if H2O_ORIGINAL_JAR is not set.
+
+User can also use gradle command line argument -PdownloadAndExtend which will download correct h2o version for current
+sparkling water from our repository if it hasn't been already downloaded.
+If this is run without any argument, then basic h2o jar is downloaded,
+but this command takes also argument specifying hadoop version, such as cdh5.4. In case the argument is specified, instead
+of downloading h2o jar this command will download h2o driver jar for specified hadoop version.
+
+Gradle property has higher priority if both H2O_ORIGINAL_JAR and -PdownloadAndExtend are set
+
+Few examples:
+
+1)
+export H2O_ORIGINAL_JAR = ...
+./gradlew build -x check
+
+In this case the jar to be extended is located using the provided environment variable
+
+2)
+./gradlew build -x check -PdownloadAndExtend
+
+In this case the jar to be extended is h2o jar and is downloaded from our repository first
+
+3)
+./gradlew build -x check -PdownloadAndExtend=cdh5.4
+
+In this case the jar to be extended is h2o driver jar for provided hadoop version and is downloaded from our repository first
+
+4)
+./gradlew build -x check -PdownloadAndExtend=abc
+
+This case will throw an exception since such hadoop version is not supported
+
+5)
+export H2O_ORIGINAL_JAR = ...
+./gradlew build -x check -PdownloadAndExtend
+
+This version will ignore environment variable and jar to be extended will be downloaded from our repository. The same
+holds for version with argument.
+"""
+
+configurations {
+    artifactsToMerge
+}
+
+def originalJar = getOrDownloadOriginalJar()
```

Comment:
What do you mean please ? I had a look and I'm not putting it in `ext` manually

I need to keep the `originalJar` at this scope for example because of this call
```
if (originalJar != null) {
    build.dependsOn extendJar
}
```